File: Jacob/lga11/src/SSQ/SSQ.java
package SSQ;

import java.util.*;

public class SSQ{

	private Queue<Event> job_queue; 

	public SSQ() {
		this.job_queue = new LinkedList<Event>();
	}
	
	public void enque_e(Event e) {
		this.job_queue.add(e);
	}
	
	public Event get_event() {
		return this.job_queue.remove();
	}
	/*
	 * l returns the l statistic
	 */
	public int get_l() {
		return this.job_queue.size();
	}
	
}

File: Jacob/lga12/src/SSQ.cpp
#include "SSQ.hpp"

std::queue<Event> job_queue; 

SSQ::SSQ(int num_servers) {
	job_queue = std::queue<Event>();
	inService = false;
	this->num_servers = num_servers;
	for (int i = 0; i < num_servers; i++){
		ServiceNode node;
		nodes.push_back(node);
	}
}
	
void SSQ::enque_e(Event e) {
	job_queue.push(e);
}

Event SSQ::get_event() {
	Event e = job_queue.front();
	job_queue.pop();
	return e;
}

void SSQ::pop_e() {
	job_queue.pop();
}

/*
* l returns the l statistic
*/
int SSQ::get_l() {
	return job_queue.size();
}

bool SSQ::is_in_service() {
	return inService;
}

void SSQ::update_service(bool u) {
	inService = u;
}

int SSQ::get_num_servers(){
	return num_servers;
}